test/e2e/cluster_samples_operator_test.go

```diff
@@ -33,24 +48,383 @@ func TestClusterSamplesOperator(t *testing.T) {
+	}
+
+	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := sdk.Get(sr); err != nil {
+			return false, nil
+		}
+		if sr.CreationTimestamp != oldTime {
+			return false, fmt.Errorf("SamplesResource was recreated when it should not have been: old create time %v new create time %v", oldTime, sr.CreationTimestamp)
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+
+	isl := &imageapiv1.ImageStreamList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ImageStreamList",
+			APIVersion: imageapiv1.SchemeGroupVersion.String(),
+		},
+	}
+	sdk.List("openshift", isl, sdk.WithListOptions(&metav1.ListOptions{}))
+	if len(isl.Items) > 0 {
+		t.Errorf("still imagestreams in openshift namespace %#v", isl.Items)
+	}
+	tl := &templatev1.TemplateList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "TemplateList",
+			APIVersion: templatev1.SchemeGroupVersion.String(),
+		},
+	}
+	sdk.List("openshift", tl, sdk.WithListOptions(&metav1.ListOptions{}))
+	if len(tl.Items) > 0 {
+		t.Errorf("still imagestreams in openshift namespace %#v", tl.Items)
+	}
 
-	istag := &imageapiv1.ImageStreamTag{
+	sr = verifyOperatorUp(t)
+	sr.Spec.ManagementState = operatorsv1alpha1api.Managed
+	err = sdk.Update(sr)
+	if err != nil {
+		t.Errorf("error updating samplesresource %v", err)
+	}
+
+	err = verifyConditionsCompleteSamplesAdded(sr)
+	if err != nil {
+		t.Errorf("samples not re-established when set to managed %#v", sr)
+	}
+
+	validateContent(t, &now)
+
+	sr = verifyOperatorUp(t)
+	sr.Spec.ManagementState = operatorsv1alpha1api.Unmanaged
+	err = sdk.Update(sr)
+	if err != nil {
+		t.Errorf("error updating samplesresource %v", err)
+	}
+
+	is := &imageapiv1.ImageStream{
 		TypeMeta: metav1.TypeMeta{
-			Kind:       "ImageStreamTag",
+			Kind:       "ImageStream",
 			APIVersion: imageapiv1.SchemeGroupVersion.String(),
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "ruby:2.4",
+			Name:      "jenkins",
 			Namespace: "openshift",
 		},
 	}
-	err = wait.PollImmediate(1*time.Second, 10*time.Minute, func() (bool, error) {
-		if err := sdk.Get(istag); err != nil {
-			return false, nil
+	err = sdk.Delete(is, sdk.WithDeleteOptions(&metav1.DeleteOptions{}))
+	if err != nil {
+		t.Errorf("error deleting jenkins imagestream %v", err)
+	}
+	// make sure jenkins imagestream does not appear while unmanaged
+	// first, wait sufficiently to make sure delete has gone though
+	err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		err := sdk.Get(is)
+		if kerrors.IsNotFound(err) {
+			return true, nil
 		}
-		return true, nil
+		return false, nil
+	})
+	if err != nil {
+		t.Errorf("delete did not occur %v", err)
+	}
+	// now make sure it has not been recreated
+	err = wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		err := sdk.Get(is)
+		if err == nil {
+			return false, fmt.Errorf("jenkins imagestream recreated")
+		}
+		return false, nil
+	})
```

Comment:
if your goal is to wait 30s and see if the imagestream got recreated, you might as well just do a time.Sleep(30*time.Second) and then check if it exists.

the wait.Poll isn't buying you anything if you're never going to exit early.

Reply:
yep that is the goal ... will switch